ocaml/parsing/parsetree.mli

```diff
@@ -431,24 +431,21 @@ and expression_desc =
   | Pexp_extension of extension  (** [[%id]] *)
   | Pexp_unreachable  (** [.] *)
 
+(** (Jane Street specific; delete when upstreaming.)
+    Consider using [Jane_syntax.Case.of_ast] before matching on this type
+    directly, as the former will detect extension nodes correctly.
+    Our syntax extensions are encoded as
+    [{ c_rhs = { pexp_desc = Pexp_match _; pexp_attributes = _attr :: _ } }]
+    if your pattern match avoids matching that pattern, it is OK to skip
+    [of_ast].
```

Comment:
I don't like this advice &mdash; let's try to keep details of the embedding outside of this file. More generally, I'd prefer to just call `of_ast` everywhere because it's pretty simple and correct.

  1. Can we delete this part of the comment?
  2. Can we change any place that currently takes this advice to instead call `of_ast`?

Reply:
I don't believe there is any place in the code that takes this advice, but it is the advice given when mentioning the other Jane_syntax syntactic categories. I agree that it is better to always call `of_ast`, so I deleted this part of the comment.